lms/djangoapps/api_manager/management/commands/bulk_delete_courses_with_reference_data.py

```diff
@@ -0,0 +1,110 @@
+    help = 'Deletes courses based on their age and relevant data of each course'
+    option_list = BaseCommand.option_list + (
+        make_option(
+            "-c",
+            "--age",
+            dest="age",
+            help="Age in days",
+            metavar="60"
+        ),
+    )
+    module_store = modulestore()
+    total_courses = 0
+    total_deleted = 0
+
+    @staticmethod
+    def delete_reference_data(course_key):
+        """
+        Deletes reference data for course
+        """
+        print 'removing reference data for course %s' % unicode(course_key)
+        CourseGroupRelationship.objects.filter(course_id=course_key).delete()
+        CourseContentGroupRelationship.objects.filter(course_id=course_key).delete()
+        CourseCohortsSettings.objects.filter(course_id=course_key).delete()
+        CourseUserGroup.objects.filter(course_id=course_key).delete()
+        CourseAggregatedMetaData.objects.filter(id=course_key).delete()
+        CourseStructure.objects.filter(course_id=course_key).delete()
+        CourseOverview.objects.filter(id=course_key).delete()
+        StudentModule.objects.filter(course_id=course_key).delete()
+        CourseModuleCompletion.objects.filter(course_id=course_key).delete()
+        StudentProgressHistory.objects.filter(course_id=course_key).delete()
+        StudentProgress.objects.filter(course_id=course_key).delete()
+        StudentGradebookHistory.objects.filter(course_id=course_key).delete()
+        StudentGradebook.objects.filter(course_id=course_key).delete()
+        CourseAccessRole.objects.filter(course_id=course_key).delete()
+        CourseEnrollment.objects.filter(course_id=course_key).delete()
+
+    def delete_course_from_modulestore(self, course_key):
+        """
+        Deletes course from modulestore
+        """
+        user_id = ModuleStoreEnum.UserID.mgmt_command
+        with self.module_store.bulk_operations(course_key):
+            print 'Removing course %s from modulestore' % unicode(course_key)
+            self.module_store.delete_course(course_key, user_id)
+
+    @transaction.commit_on_success
+    def delete_course_and_data(self, course_key):
+        """
+        Delete course from modulestore and it reference data
+        """
+        Command.delete_reference_data(course_key)
+        self.delete_course_from_modulestore(course_key)
+
+    def handle(self, *args, **options):
+        if not options.get('age'):
+            raise CommandError("bulk_delete_courses_with_reference_data command requires one integer argument: --age")
+
+        age_in_days = int(options.get('age'))
+        if query_yes_no(
+                "Are you sure you want to delete all courses and their reference data having no activity "
+                "in last %s days. This action cannot be undone!" % age_in_days, default="no"
+        ):
+            created_time = datetime.now(pytz.UTC) + timedelta(days=-age_in_days)
+            courses = self.module_store.get_courses()
```

Comment:
I would do this before confirming with the user that they want to proceed and include the number of courses that are going to be deleted in the confirmation prompt.